GNUmakefile

```diff
@@ -58,30 +56,50 @@ SHELL-NAME := makes go-yaml
 include $(MAKES)/clean.mk
 include $(MAKES)/shell.mk
 
-MAKES-CLEAN += $(dir $(YTS-DIR)) $(GOLANGCI-LINT)
+MAKES-CLEAN := $(CLI-BINARY) $(GOLANGCI-LINT)
+MAKES-REALCLEAN := $(dir $(YTS-DIR))
+
+SHELL-SCRIPTS := \
+  util/common.bash \
+  $(shell grep -rl '^#!/usr/bin/env bash' . | \
+          grep -Ev '(\.sw|\.cache/)')
 
+# v=1 for verbose
 v ?=
+# o='--cover' for options
+o ?=
 count ?= 1
 
 
 # Test rules:
-test: $(GO-DEPS)
-	go test$(if $v, -v) -vet=off --cover . ./internal/...
+test: test-cover test-yts-all test-shell
+	@echo 'ALL TESTS PASS'
+
+test-unit: $(GO-DEPS)
+	go test$(if $v, -v) .$(if $o, $o)
 
-test-data: $(YTS-DIR)
+test-internal: $(GO-DEPS)
+	go test$(if $v, -v) ./internal/...$(if $o, $o)
 
-test-all: test test-yts-all
+test-cover: $(GO-DEPS)
+	go test$(if $v, -v) . ./internal/... -vet=off --cover$(if $o, $o)
 
 test-yts: $(GO-DEPS) $(YTS-DIR)
-	go test$(if $v, -v) ./yts -count=$(count)
+	go test$(if $v, -v) ./yts$(if $o, $o)
 
 test-yts-all: $(GO-DEPS) $(YTS-DIR)
 	@echo 'Testing yaml-test-suite'
-	@RUNALL=1 bash -c "$$yts_pass_fail"
+	util/yaml-test-suite all
 
 test-yts-fail: $(GO-DEPS) $(YTS-DIR)
 	@echo 'Testing yaml-test-suite failures'
-	@RUNFAILING=1 bash -c "$$yts_pass_fail"
+	util/yaml-test-suite fail
+
+test-shell: $(SHELLCHECK)
```

Comment:
FYI, I had to upgrade my local `make` installation as this was failing on 3.81. Once I upgraded to 4.4.1 it worked as expected. It might be worth listing a minimum `make` version somewhere, e.g. in the readme.